src/modules/wallet/Amount.ts

```diff
@@ -0,0 +1,15 @@
+import { Currency } from '../GameConstants';
+
+export default class Amount {
+    amount: number;
+    currency: Currency;
+
+    constructor(amount: number, currency: Currency) {
```

Comment:
While you have your finger in this particular pie...

Notice how we have amount and currency written out three times each? It turns out there is a neater way to do this.
You can turn
```typescript
class Amount {
  amount: number;
  currency: Currency;
  
  constructor(amount: number, currency: Currency) {
    this.amount = amount;
    this.currency = currency
  }
}
```
into the much more concise:
```ts
class Amount {
  constructor(
    public amount: number,
    public currency: Currency
  ) { }
}
```

Reply:
That's really dope. Is there a place I can go to learn more about refactoring? The guides I've used aren't that helpful with this game. Also, I really do appreciate you taking the time to teach me this stuff. I've learned more on this game than I did in school. :/

Reply:
For OOP stuff I would look into SOLID principles - it's not exactly about refactoring, but more about designing things in a less entangled mess, which is the direction a refactor should probably be taking you. I don't have a good guide on those things, but if you google around you might find something that makes sense to you.

This particular thing I think I came across while reading typescript documentation, I didn't know I wanted it until I saw it done :sweat_smile: